letsencrypt/models/base_config_settings.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class BaseConfigSettings(models.TransientModel):
+    _inherit = 'base.config.settings'
+
+    letsencrypt_dns_provider = fields.Selection(
+        [('shell', 'Shell')],
+        string='DNS Provider',
+        help='Set this to Shell if you want an external script to be run.')
+    letsencrypt_shell = fields.Text()
+    letsencrypt_url = fields.Char(string='URL')
+
+    def _compute_letsencrypt_needs_dns_provider(self):
+        ir_config_parameter = self.env['ir.config_parameter']
+        altnames = ir_config_parameter.search([
+            ('key', 'like', 'letsencrypt.altname.%')])
+        for rec in self:
+            for name in altnames:
+                if '*' in name:
```

Comment:
why don't you just add this to your search? and then counting the results would be sufficient, no need to actually fetch the records